api/apiclient.go

```diff
@@ -462,9 +467,15 @@ func apiPath(model, path string) (string, error) {
 // dialResult holds a dialed connection, the URL
 // and TLS configuration used to connect to it.
 type dialResult struct {
-	conn      jsoncodec.JSONConn
-	addr      string
-	urlStr    string
+	conn jsoncodec.JSONConn
+	// controllerRootAddr represents the controller's root address.
+	// E.g. wss://controller.com/foo
+	controllerRootAddr *url.URL
+	// dialAddr is the full URL (with no DNS resolution) that was actually dialed
+	// e.g. wss://controller.com/foo/api for the controller endpoint
```

Comment:
Make the "e.g." here consistent

Also for the controllerRootAddr and dialAddr examples, can we make one the FQDN and the other the IP address to show the possible difference.